File: qc-core/src/main/java/com/eager/questioncloud/core/domain/review/implement/QuestionReviewStatisticsGenerator.java
package com.eager.questioncloud.core.domain.review.implement;

import com.eager.questioncloud.core.domain.question.event.RegisteredQuestionEvent;
import com.eager.questioncloud.core.domain.review.infrastructure.QuestionReviewStatisticsRepository;
import com.eager.questioncloud.core.domain.review.model.QuestionReviewStatistics;
import lombok.RequiredArgsConstructor;
import org.springframework.context.event.EventListener;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class QuestionReviewStatisticsGenerator {
    private final QuestionReviewStatisticsRepository questionReviewStatisticsRepository;

    @EventListener
    public void generate(RegisteredQuestionEvent event) {
        questionReviewStatisticsRepository.save(QuestionReviewStatistics.create(event.getQuestion().getId()));
    }
}

File: qc-core/src/main/java/com/eager/questioncloud/core/domain/review/implement/QuestionReviewStatisticsGenerator.kt
package com.eager.questioncloud.core.domain.review.implement

import com.eager.questioncloud.core.domain.review.infrastructure.repository.QuestionReviewStatisticsRepository
import com.eager.questioncloud.core.domain.review.model.QuestionReviewStatistics.Companion.create
import org.springframework.stereotype.Component

@Component
class QuestionReviewStatisticsGenerator(
    private val questionReviewStatisticsRepository: QuestionReviewStatisticsRepository
) {
    fun generate(questionId: Long) {
        questionReviewStatisticsRepository.save(create(questionId))
    }
}
